eldritch/events.py

```diff
@@ -2051,16 +2122,19 @@ def resolve_internal(self, choices):
 
 class ItemCountChoice(ItemChoice):
 
-  def __init__(self, character, prompt, count):
-    super(ItemCountChoice, self).__init__(character, prompt)
+  def __init__(self, character, prompt, count, min_count=None, decks=None):
+    super(ItemCountChoice, self).__init__(character, prompt, decks=decks)
     self.count = count
+    self.min_count = count if min_count is None else min_count
 
   def resolve_internal(self, choices):
-    assert self.count == len(choices)
+    assert self.min_count <= len(choices) <= self.count
     super(ItemCountChoice, self).resolve_internal(choices)
 
 
 class SinglePhysicalWeaponChoice(ItemChoice):
+  # def __init__(self, char, prompt):
```

Comment:
You could make this inherit from ItemCountChoice and override the constructor to supply count and min_count to the ItemCountChoice constructor. Then you could get rid of resolve_internal